### ISSUE TITLE
[BUG]: Lock / unlock from HA?

### ISSUE THREAD

**Issue by sammyke007** [CoMPaTech/stromer#26]
### Describe the bug.

Hi

At first I would like to thank you for this great HA integration.

Is there any plan to make it possible to lock / unlock the bike using a HA switch?
If I understand correctly it only reads data from the API now.

Kind regards
Sammy

### What version of Home Assistant Core has the issue?

2023.5.4

### What was the last working version of Home Assistant Core?

2023.5.4

### What type of installation are you running?

Home Assistant OS

### How did you install the stromer integration

HACS

### Logging

/

### Diagnostics information

/

### Additional information

/

**Comment by CoMPaTech** [comment]
Hi @sammyke007 is this still on the wish-list? I might be able to put some effort in to try if we can tinker with small things using this integration? (As in, might be able to squeeze some time into the upcoming holiday season)

**Comment by sammyke007** [comment]
@CoMPaTech , yes that would be nice.

I already use it now with the following method:

**configuration.yaml:**
```
python_script:
  requirements:
  - setuptools==67.2.0
  - requests>=2.28.2
  - XlsxWriter~=3.0.8
  - stromer_api
 ```

**In /config/scripts/stromerunlock.py:**
```
from stromer_api import StromerBike 
mybike = StromerBike("REDACTED@gmail.com", "REDACTED_PASS", "REDACTED_CLIENT_ID")
# access available data easily
mybike.unlock()
```

and then I call the python script with a HA script...
```
alias: Stromer unlock
sequence:
  - service: python_script.exec
    data:
      file: scripts/stromerunlock.py
      cache: false
mode: single
icon: mdi:bike-fast
```
So it's a little bit of a hassle, but it works very good.

**Comment by CoMPaTech** [comment]
It definitely needs some polishing (see the screenshot in #72) but initially looks to be working fine (obviously can't turn the light on with a locked bike). Needs some more work to properly get the current state in, but it's sleepy time now. Should be on the right track.
And I know you didn't ask for light, but that's far more convenient than refreshing the app or physically inspect the bike :)

**Comment by CoMPaTech** [comment]
Hi @sammyke007 - I think https://github.com/CoMPaTech/stromer/releases/tag/v0.3.0b1 should be relatively safe as a beta for testing. Do note the 'bouncing' issue; checking bearings to see how to (more) properly resolve that. As it is now, HA will 'honor' you request to switch on/off, (meanwhile re-requesting API status for everything), than conclude that the known state was as before (as the API is quiet slow), and then update to API status. I know I can just 'override' whatever is in the cached API status but that is a thing we can fix in another beta version.

You should be able to deploy b1 (in HACS, go to the Stromer tile, hit the three dots and 'update information' and 'download' where on that one you will have the 'show beta' releases). Please do not install any of the alphas 😄

**Comment by sammyke007** [comment]
I'll try it in a moment!

**Comment by sammyke007** [comment]
Locking and unlocking works surprisingly fast. The light switch also works for me, but instead off on and off, it toggles between normal and bright mode (dim and far). I have an ST3 Pinion from 2023 and disabling the lights is impossible. Just FYI

**Comment by CoMPaTech** [comment]
For me locking was 'about as fast' as using the app, just that the switch sometime bounces visibly in HA, but nothing I would worry about to much (just aesthetics). On my '2' I can off/on ... according to the library it could also 'flash' but didn't tinker with that. I'll update the PR in a while to amend that it's switching 'light-mode' :) That might entail all options (i.e. full on vs either dim or off). Thanks for testing.

**Comment by CoMPaTech** [comment]
Released as v0.3.1 (though now called light-mode unfortunately HA doesn't update when it's already set. So manually rename them if you like)

**Comment by sammyke007** [comment]
On my previous "2" it was indeed light on/off. It toggles the same option as pressing the light button on your steering wheel control.

ST2: toggle ON/OFF
ST3 Pinion: DIM/BRIGHT
